packages/cli/src/tools/merge.ts

```diff
@@ -0,0 +1,15 @@
+import deepmerge from 'deepmerge';
+
+/**
+ * `deepmerge` concatenates arrays by default instead of overwriting them.
+ * We define custom merging function for arrays to change that behaviour
+ */
+export default function merge(
```

Comment:
There should be a better way to `infer` those parameters. For now, I think it is okay to leave it because it might cause errors in other parts (especially those which are not migrated to TS yet).